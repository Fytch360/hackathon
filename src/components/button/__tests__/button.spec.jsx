import React from 'react'
import { shallow, mount } from 'enzyme'
import { matchers } from '@emotion/jest'

import { ButtonSecondary, ButtonTertiary, ButtonTransparent, ButtonPrimary } from '..'

expect.extend(matchers)

describe('<Button />', () => {
    test('is available', () => {
        expect(ButtonPrimary).toBeDefined()
        expect(ButtonSecondary).toBeDefined()
        expect(ButtonTertiary).toBeDefined()
        expect(ButtonTransparent).toBeDefined()
    })

    test('renders title', () => {
        const wrapper = shallow(<ButtonSecondary title="title" />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.text('title')).toBeTruthy()
    })

    test('fire onClick handler', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title'
        }
        const wrapper = shallow(<ButtonPrimary {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.simulate('click')
        expect(clickHandler).toHaveBeenCalled()
    })

    test('does not fire onClick handler if disabled', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title',
            disabled: true
        }
        const wrapper = mount(<ButtonPrimary {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.simulate('click')
        expect(clickHandler).not.toHaveBeenCalled()
    })

    test('renders loading', () => {
        const wrapper = shallow(<ButtonPrimary title="title" />)
        expect(wrapper.props().loading).not.toBe(true)
        const wrapper2 = shallow(<ButtonPrimary title="title" loading />)
        expect(wrapper2.props().isLoading).toBe(true)
    })

    test('does not fire onClick handler if loading', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title',
            loading: true
        }
        const wrapper = mount(<ButtonPrimary {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.simulate('click')
        expect(clickHandler).not.toHaveBeenCalled()
    })

    test('does disabled if loading', () => {
        const wrapper = mount(<ButtonPrimary title="title" loading disabled />)
        expect(wrapper.props().disabled).toBe(true)
    })

})
