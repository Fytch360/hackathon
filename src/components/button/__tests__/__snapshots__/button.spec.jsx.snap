// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders title 1`] = `
<ButtonSecondaryStyled
  aria-busy={false}
  aria-label="title"
  aria-live="polite"
  iconReverse={false}
  isLoading={false}
  onClick={[Function]}
  size="md"
>
  <ButtonBase
    fontWeight="medium"
    iconReverse={false}
    size="md"
    title="title"
  />
</ButtonSecondaryStyled>
`;
